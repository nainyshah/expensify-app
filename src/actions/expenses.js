import { v4 as uuidv4 } from 'uuid';

// ==========Expense Actions =============
export const addExpense = ({
	description = '',
	note = '',
	amount = 0,
	createdAt = 0,
} = {}) => ({
	type: 'ADD_EXPENSE',
	expense: {
		id: uuidv4(),
		description,
		note,
		amount,
		createdAt,
	},
});
export const removeExpense = ({ id } = {}) => ({
	type: 'REMOVE_EXPENSE',
	id,
});

export const editExpense = (id, updates) => ({
	type: 'EDIT_EXPENSE',
	id,
	updates,
});

// export const addExpense = ({
// 	type: 'ADD_EXPENSE',
// 	expense
// });

// export const startAddExpense = (expenseData = {}) => {
// 	return (dispatch) => {
// 		const {
// 			description = '',
// 			note = '',
// 			amount = 0,
// 			createdAt = 0,
// 		} = expenseData;
// 		const expense = { description, note, amount, createdAt };
// 		database
// 			.ref('expense')
// 			.push(expense)
// 			.then(() => {
// 				addExpense({
// 					id: ref.key,
// 					...expense,
// 				});
// 			});
// 	};
// };
