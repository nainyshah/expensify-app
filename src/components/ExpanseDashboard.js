import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilter from './ExpenseListFilters';

const ExpnaseDashboardPage = () => (
	<div>
		<ExpenseListFilter />
		<ExpenseList />
	</div>
);

export default ExpnaseDashboardPage;
