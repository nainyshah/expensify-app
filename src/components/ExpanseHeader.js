import React from 'react';
import { NavLink } from 'react-router-dom';

const ExpanseHeader = () => (
	<header>
		<h1>Expensify</h1>
		<p>
			<NavLink activeClassName="is-active" to="/" exact={true}>
				Home
			</NavLink>
			|
			<NavLink activeClassName="is-active" to="/create">
				Create
			</NavLink>
			|
			<NavLink activeClassName="is-active" to="/help">
				Help
			</NavLink>
		</p>
	</header>
);

export default ExpanseHeader;
