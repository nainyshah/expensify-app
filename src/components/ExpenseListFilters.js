import React from 'react';
import { connect } from 'react-redux';
import {
	setTextFilter,
	setSortByAmount,
	setSortByDate,
	setStartDate,
	setEndDate,
} from '../actions/filters';

import { DateRangePicker } from 'react-dates';

export class ExpenseListFilters extends React.Component {
	state = {
		calendarFocused: null,
	};
	onDatesChange = ({ startDate, endDate }) => {
		this.props.setStartDate(startDate);
		this.props.setEndDate(endDate);
	};
	onFocusChange = (calendarFocused) => {
		this.setState(() => ({ calendarFocused }));
	};
	onTextChange = (e) => {
		this.props.setTextFilter(e.target.value);
	};
	onSortChange = (e) => {
		if (e.target.value === 'amount') {
			this.props.setSortByAmount();
		} else if (e.target.value === 'date') {
			this.props.setSortByDate();
		}
	};
	render() {
		return (
			<div>
				<input
					type="text"
					value={this.props.filters.text}
					onChange={this.onTextChange}
				></input>
				<select value={this.props.filters.sortBy} onChange={this.onSortChange}>
					<option value="date">Date</option>
					<option value="amount">Amount</option>
				</select>
				<DateRangePicker
					startDateId="MyDatePicker"
					endDateId="MyDatePicker"
					startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
					endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
					onDatesChange={this.onDatesChange}
					focusedInput={this.state.calendarFocused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
					onFocusChange={this.onFocusChange} // PropTypes.func.isRequired,
					showClearDates={true}
					numberOfMonths={1}
					isOutsideRange={() => false}
				/>
			</div>
		);
	}
}

const mapDispatchToProps = (dispatch, props) => ({
	setStartDate: (startDate) => dispatch(setStartDate(startDate)),
	setEndDate: (endDate) => dispatch(setEndDate(endDate)),
	setTextFilter: (text) => dispatch(setTextFilter(text)),
	setSortByAmount: () => dispatch(setSortByAmount()),
	setSortByDate: () => dispatch(setSortByDate()),
});

const mapStateToProps = (state) => {
	return {
		filters: state.filters,
	};
};
export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
