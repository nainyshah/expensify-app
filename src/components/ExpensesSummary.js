import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expense-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
	const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
	const formattedExpenseTotal = numeral(expensesTotal / 100).format('$0,0.00');
	return (
		<h3>
			Viewing {expenseCount} {expenseWord} totalling {formattedExpenseTotal}
		</h3>
	);
};

// export default ExpensesSummary;
const mapStateToProps = (state) => {
	const visibleExpenses = selectExpenses(state.expenses, state.filters);
	return {
		expenseCount: visibleExpenses.length,
		expensesTotal: selectExpensesTotal(visibleExpenses),
	};
};

export default connect(mapStateToProps)(ExpensesSummary);
