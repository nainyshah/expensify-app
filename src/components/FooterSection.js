import React from 'react';

const FooterSection = (props) => (
  <section id="footer-section" className="bg-primary fixed-bottom">
    <div className="container">
      <div className="row">
        <div className="col text-center py-2">
          <h3>{props.footerTitle}</h3>
          <p>{props.copyRightText}</p>
        </div>
      </div>
    </div>
  </section>
);

export default FooterSection;
