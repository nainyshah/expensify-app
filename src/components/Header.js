import React from 'react';
import Navbar from 'react-bootstrap/Navbar';
const Header = (props) => (
  <nav
    id="main-nav"
    className="navbar navbar-expand-sm bg-dark navbar-dark fixed-top"
  >
    <div className="container">
      <a href="#" className="navbar-brand">
        {props.title}
      </a>
    </div>
  </nav>
);

Header.defaultProps = {
  title: 'Syed Raza',
};

export default Header;
