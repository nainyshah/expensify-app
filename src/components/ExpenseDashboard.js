import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilter from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const ExpneseDashboard = () => (
	<div>
		<ExpensesSummary />
		<ExpenseListFilter />
		<ExpenseList />
	</div>
);

export default ExpneseDashboard;
