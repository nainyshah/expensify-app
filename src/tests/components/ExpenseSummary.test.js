import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import selectExpenseTotal from '../../selectors/expense-total';
import expenses from '../fixtures/expenses';

test('should correctly render ExpenseSummary with 1 expense', () => {
	const wrapper = shallow(
		<ExpensesSummary expenseCount={1} expensesTotal={expenses[0].amount} />
	);
	expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpenseSummary with all expense', () => {
	const wrapper = shallow(
		<ExpensesSummary
			expenseCount={expenses.length}
			expensesTotal={selectExpenseTotal(expenses)}
		/>
	);
	expect(wrapper).toMatchSnapshot();
});
