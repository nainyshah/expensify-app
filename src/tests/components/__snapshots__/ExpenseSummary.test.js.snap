// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with 1 expense 1`] = `
<h3>
  Viewing 
  1
   
  expense
   totalling 
  $5.00
</h3>
`;

exports[`should correctly render ExpenseSummary with all expense 1`] = `
<h3>
  Viewing 
  3
   
  expenses
   totalling 
  $44.00
</h3>
`;
